### DISCUSSION TITLE
Mini.test get value from "vim.fn.getreg('*')" ?

### DISCUSSION THREAD

**Discussion by xixiaofinland** [nvim-mini/mini.nvim#1092] [Q&A] [answered]
I understand many APIs in `fn` has been exposed to `child` so `vim.fn` equals to `child.fn`.
But both the last 2 lines retrieve value from the current Nvim running instance (they one I'm editing the test file) rather than the child. Why is it so while the `vim.bo` ones work as expected?

```lua
T['copy_apex_name()']['return apex name'] = function()
  local apex_name = 'SfProject'
  local lua_cmd = string.format('edit tests/dir/sf-project/%s.cls', apex_name)
  child.cmd(lua_cmd)

  child.type_keys('<leader>cc') -- copy name 'SfProject' into "*" registration

  eq(child.lua_get('vim.bo.filetype'), 'apex')  -- works
  eq(child.bo.filetype, 'apex')  -- works

  eq(child.lua_get('vim.fn.getreg("*")'), apex_name)  -- it gets the registration value from the current Nvim instance rather than the test child.
  eq(child.fn.getreg("*"), apex_name) -- same as above line
end

end
```

**Comment by xixiaofinland** [comment]
So the `Yanked_value` below is a current yanked value in my running Nvim instance.

Test result:
```
Fails (1) and Notes (0)
FAIL in tests/test_apis.lua | copy_apex_name() | return apex name:
  Failed expectation for equality.
  Left:  "Yanked_value"
  Right: "SfProject"
  Traceback:
    tests/test_apis.lua:69

make: *** [Makefile:3: test] Error 1
```

**Comment by echasnovski** [comment, marked as answer]
Unfortunately, working with system register is indeed a bit messy. The reason (I think) is mostly because it is something shared by both instances and that Neovim has special treatment of system clipboard (see `:h clipboard`).

Here is [one example of how I get over it](https://github.com/echasnovski/mini.nvim/blob/59ecf78491f29a9ec31f4124f3da77091f37e4d9/tests/test_extra.lua#L2841-L2851).

------

> ```
> ...
> child.type_keys('<leader>cc') -- copy name 'SfProject' into "*" registration
> ...
> ```

The answer would have been more helpful if you posted a fully reproducible (and small-ish) test case. For example, the following test case passes locally for me:

```lua
T['tmp'] = function()
  child.fn.setreg('*', 'Hello')
  eq(child.fn.getreg('*'), 'Hello')
end
```

**Reply by xixiaofinland** [reply]
Interesting, your 'temp' indeed passes. Here is the function it runs with `<leader>cc`, it should be functional-wise identical to your passed test case.

Meanwhile, I'm debugging it. When I invoke the function `child.lua([[Util.copy_apex_name()]])` instead of user typing, it works as expected.

```lua
-- Copy current file name without dot-after, e.g. copy "Hello" from "Hello.cls"
M.copy_apex_name = function()
  local file_name = vim.split(vim.fn.expand("%:t"), ".", { trimempty = true, plain = true })[1]
  vim.fn.setreg('*', file_name)
  vim.notify(string.format('"%s" copied.', file_name), vim.log.levels.INFO)
end
```
